src/nunit.analyzers/ClassicModelAssertUsage/ClassicModelAssertUsageCodeFix.cs

```diff
@@ -52,30 +54,65 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             if (newInvocationNode is null)
                 return;
 
-            // Now, replace the arguments.
-            List<ArgumentSyntax> arguments = invocationNode.ArgumentList.Arguments.ToList();
+            var methodSymbol = (IMethodSymbol)semanticModel.GetSymbolInfo(invocationNode).Symbol!;
 
-            // See if we need to cast the arguments when they were using a specific classic overload.
-            arguments[0] = CastIfNecessary(arguments[0]);
-            if (arguments.Count > 1)
-                arguments[1] = CastIfNecessary(arguments[1]);
+            Dictionary<string, ArgumentSyntax> argumentNamesToArguments = new();
+            const string NameOfArgsParameter = "args";
+
+            // There can be 0 to any number of message and parameters.
+            List<ArgumentSyntax> messageAndParams = new();
+            var arguments = invocationNode.ArgumentList.Arguments.ToList();
+            for (var i = 0; i < arguments.Count; ++i)
+            {
+                var argument = arguments[i];
+                if (i < methodSymbol.Parameters.Length
+                    && (argument.NameColon?.Name.Identifier.Text ?? methodSymbol.Parameters[i].Name) is { } argumentName
```

Comment:
```suggestion
                    && (argument.NameColon?.Name.Identifier.Text ?? methodSymbol.Parameters[i].Name) is string argumentName
```

I find the `is { }` not very readable. Specifying the expected type make it clearer.